torch/fx/experimental/proxy_tensor.py

```diff
@@ -657,31 +657,50 @@ def disable_autocast_cache():
         torch.set_autocast_cache_enabled(old_value)
 
 
-def make_fx(f, decomposition_table=None, tracing_mode="real", _allow_non_fake_inputs=False):
+def make_fx(f, decomposition_table=None, tracing_mode="real", *, _allow_non_fake_inputs=False, _example_inputs=None):
```

Comment:
Adding `_example_inputs` would seem to create a kind of awkward usage:
```
gm = make_fx(fn, _example_inputs=[a,b,c])(a,b,c)
``` 
Does make_fx even need the non-fake inputs?

Reply:
Agreed, although I see the technical problem that resulted in this. Thinking...

Reply:
OK, so, if you do this exactly the way AOTAutograd does it, you don't need to take in `example_inputs` in `make_fx`, because you are going to call it with `tracing_mode="real"`, and the calling code is going to take care of all the ShapeEnv creation, fake tensor mode entering, and input translation.

So perhaps the best way to do this, is to make a wrapper around `make_fx`, that calls `make_fx` in the same way, but does all this stuff. The reason we can't make `make_fx` do this is because it's staged so that you don't take the inputs immediately, and this is incompatible with any API style where we work things out from the inputs.

Reply:
Actually it's not even clear to me how much of a wrapper you even need, given that you should already have fake tensors ambiently available. Just call `tracing_mode="real"`! Do you even need this PR? Try the E2E.

Reply:
Yeah I think i got it working with just real mode.